packages/itwinui-react/src/core/TransferList/TransferList.tsx

```diff
@@ -0,0 +1,275 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  getFocusableElements,
+  useMergedRefs,
+  useTheme,
+} from '../utils/index.js';
+import type {
+  CommonProps,
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../utils/index.js';
+
+import '@itwin/itwinui-css/css/transfer-list.css';
+import { List, ListItem } from '../List/index.js';
+import type { ListItemProps, ListProps } from '../List/index.js';
+// ----------------------------------------------------------------------------
+// TransferList.Area component
+
+type TransferListAreaOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+export type TransferListAreaProps<T extends React.ElementType = 'div'> =
+  PolymorphicComponentProps<T, TransferListAreaOwnProps>;
+
+const TransferListArea = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', children, className, ...rest } = props;
+
+  return (
+    <Element
+      className={cx('iui-transfer-list-area', className)}
```

Comment:
is it possible to remove this extra wrapper div?

ideally, the component would look like this:

```
<TransferList>
  <TransferList.Listbox>
   <TransferList.Item>...</TransferList.Item>
   <TransferList.Item>...</TransferList.Item>
   <TransferList.Item>...</TransferList.Item>
  </TransferList.Listbox>

  <TransferList.Toolbar>...</TransferList.Toolbar>

  <TransferList.Listbox>
   <TransferList.Item>...</TransferList.Item>
   <TransferList.Item>...</TransferList.Item>
  </TransferList.Listbox>
</TransferList>
```

Reply:
two things came to mind:
1. css grid could be used to keep the layout and get rid of the extra div
2. right now, it looks like the label is not even associated with the listbox (missing `aria-labelledby`), so maybe the `label` should be a prop on `TransferList.Listbox`.

Reply:
1. I tried using `grid-template-area` and I don't think we can get the desired layout since the grid involves more than one element with the same class name. The grid would have to be constructed like so:

`'label . label'`
`'listbox toolbar listbox'`

Correct me if I'm wrong on this since I'm not too familiar with `display: grid`

2. If `TransferList.Label` is replaced with a `label` prop, that would lead to the label not being polymorphic. Is that an issue? Can we add a second prop to change the label's element type?

Reply:
if the css grid is too complicated, then lets keep the extra divs, but we still need to associate the label using `aria-labelledby`. context seems like the right solution for this.
```jsx
// this will set id in context
<TransferList.Area>

  // this will read id from context, and set `id` attribute
  <TransferList.Label label='Blah' />

  // this will read id from context, and set `aria-labelledby`
  <TransferList.List>...</TransferList.List>
</TransferList.Area>
```

also renaming suggestions:
- TransferList.Area -> TransferList.ListboxWrapper
- TransferList.List -> TransferList.Listbox
- TransferList.Label -> TransferList.ListboxLabel
- TransferList.ListItem -> TransferList.Item

Reply:
Context is now used to pass a `labelId` to the label and listbox subcomponents

These components have also been renamed